### DISCUSSION TITLE
KE100 and KH100 - newbie question - how do I find the IP addresses so that I can add the KE100s?

### DISCUSSION THREAD

**Discussion by Blobfish83** [petretiandrea/home-assistant-tapo-p100#750] [General]
Hi, I've added the KH100 hub but none of the KE100s are appearing automatcially in HA. I'm looking to add manually, but for some reason the IP addresses of my KE100s are not exposed in either the tapo or kasa apps.

Would anyone be able to give me some advice on how to add the KE100s into HA or where to find the IP addresses? 

Many thanks
![HA KH100](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/124310819/5244e450-792d-439e-b5df-7e54a3eba40a)

**Comment by Rosko77** [comment]
Hello Blobfish83,
first about Tapo/Kasa. The KH100 is a hub. This communicates with the KE100's via a Sub-G radio signal. So no WiFi as you expect. Only the hub connects to your network via WiFi.
However, that is not the problem. I also use 2 KH100 with 7 KE100 on 2 floors.
Unfortunately it no longer works since version 3.1.0 - 3.1.2.
Please try version 3.0.0. I have no problems with this and the hub and the KE100 are found and can be used. Simply click on the 3 dots in HACS at the top right - Download again - and select and install version 3.0.0. Greetings Rosko
Translated from German to English using Google Translate

**Reply by Blobfish83** [reply]
Thanks for getting back to me on this! 
I understand about the IP now - makes sense!

I've followed your instructions and redownloaded 3.0.0 Unfortunately the KE100s have not been found automatically. 
Is there anything I have to do to add them manually?

Thanks again for your help.

**Comment by Rosko77** [comment]
After a HA restart at the latest, they should appear one after the other. Please try uninstalling the integration completely and downloading it again. Then create the hub again and re-enter your access data. I suspect a problem with the login from version 3.1.2. The hub should then be back immediately. If no further devices appear after this, add the following to the configuration.yaml: 
tapo:
  discovery: true # you can omit "discovery" by default, with "tapo:" it will enable discovery automatically.

Grüße Rosko

**Reply by Blobfish83** [reply]
So I completely deleted and I redownloaded the integration. Thenadded the HUB, no KE100s have been discovered. My other sockets etc have all been discovered. So automatic discovery is working. It's just not picking up the KE100s.

I added the 2 lines into the yaml just incase but it hasn't seemed to help. I don't suppose it could be causing issues that my hub is connected via a mesh node within my network?

**Reply by Rosko77** [reply]
Are the KE100 connected and controllable in the Kasa or Tapo APP? It may help to reconnect the KE100 to the HK100. Then restart HA. Greetings Rosko

**Reply by Blobfish83** [reply]
Yeah, they're all controllable in the kasa and tapo app. they're even available in the Alexa (echo) app.

**Reply by Blobfish83** [reply]
It sounds like lots of people are having similiar issues. It might be more straight forwards if I just wait for a week or two to see if a new version of the integration fixes things. Thanks for all your help though.

**Reply by Rosko77** [reply]
Sorry that nothing has helped so far. I also had problems with the KH100 from version 3.1.0 onwards. It was only when I went back to 3.0.0 that the KE100 were displayed again and could be controlled. I'm now hoping for 3.1.3 For Alexa integration. For me, Alexa sometimes also turned off the thermostats using a voice command when I turned off the lights. So if you notice that the radiator is no longer heating, check it out.
